src/components/Navbar.tsx

```diff
@@ -0,0 +1,183 @@
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case "Enter":
+      case " ":
+        // If is already selected, then deselect
+        if (id === selectedId) {
+          setSelectedId?.(undefined);
+        } else {
+          setSelectedId?.(id);
+          onSelect({ title, id });
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    registerItem?.(itemRef);
+  }, [registerItem]);
+
+  const isSelected = id === selectedId;
+
+  return (
+    <li
+      tabIndex={0}
+      className="navbar-item"
+      onClick={handleClick}
+      onKeyDown={handleKeyDown}
+      ref={itemRef}
+    >
+      <FontAwesomeIcon
+        className="navbar-item-icon"
+        icon={icon}
+        size="2x" // this should be a param
+        color={
+          isSelected
+            ? style.primary
+            : theme?.colorMode === "dark"
+            ? style.lightBackground
+            : style.darkBackground
+        }
+      />
+      <div className={`navbar-item-title ${isSelected ? "selected" : ""}`}>
+        <span className="navbar-item-title-text">{title}</span>
+      </div>
+    </li>
+  );
+}
+
+type NavbarProps = {
+  label: string;
+  children: React.ReactNode;
+};
+
+export function Navbar(props: NavbarProps) {
+  const [selectedId, setSelectedId] = useState<number>();
+  const [items, setItems] = useState<React.RefObject<HTMLLIElement>[]>([]);
+  const selectedIndicatorRef = useRef<HTMLDivElement>(null);
+
+  // TODO: refactor this monstrosity (maybe ask Sandrina to give some tips)
+  const animateWithCss = (
+    domMenuItem: HTMLLIElement,
+    domSelectedIndicator: HTMLDivElement
+  ) => {
+    // First time an item is selected, the icon appears
+    if (!domSelectedIndicator.style.visibility) {
+      domSelectedIndicator.style.top = `${domMenuItem.offsetHeight - 12}px`;
+      domSelectedIndicator.style.left = `${
+        // center positioning the icon
+        domMenuItem.offsetLeft +
+        domMenuItem.clientWidth / 2 -
+        domSelectedIndicator.clientWidth / 2
+      }px`;
+      domSelectedIndicator.style.visibility = "visible";
+    } else {
+      // If an item is already selected, move the icon to next position
+      const currentPosition = domSelectedIndicator.offsetLeft;
+      const nextLeft = domMenuItem.offsetLeft;
+      const movement = nextLeft - currentPosition;
+      domSelectedIndicator.style.setProperty(
+        "--translate-x-value",
+        `${
+          movement +
+          domMenuItem.clientWidth / 2 -
+          domSelectedIndicator.clientWidth / 2
+        }px`
+      );
+      domSelectedIndicator.classList.add("animate");
+    }
+  };
+
+  useLayoutEffect(() => {
+    const domMenuItem = items[selectedId!]?.current;
+    const domSelectedIndicator = selectedIndicatorRef.current;
+    if (domMenuItem && domSelectedIndicator) {
+      animateWithCss(domMenuItem, domSelectedIndicator);
```

Comment:
bug: It works well in the initial render, but if the component re-renders with just 3 or 5 items, it breaks. It's missing a dependency in one of the hooks ;)

Reply:
fixed :raised_hands: